mk/toolchain.mk

```diff
@@ -9,24 +9,14 @@ else
 endif
 
 # Validate GNU Toolchain for RISC-V
-CROSS_COMPILE ?= riscv32-unknown-elf-
-RV32_CC = $(CROSS_COMPILE)gcc
-RV32_CC := $(shell which $(RV32_CC))
-ifndef RV32_CC
-    # xPack GNU RISC-V Embedded GCC
-    CROSS_COMPILE = riscv-none-elf-
-    RV32_CC = $(CROSS_COMPILE)gcc
-    RV32_CC := $(shell which $(RV32_CC))
-    ifndef RV32_CC
-        # DEPRECATED: Replaced by xpack-dev-tools/riscv-none-elf-gcc-xpack
-        CROSS_COMPILE = riscv-none-embed-
-        RV32_CC = $(CROSS_COMPILE)gcc
-        RV32_CC := $(shell which $(RV32_CC))
-        ifndef RV32_CC
-        $(warning No GNU Toolchain for RISC-V found.)
-        CROSS_COMPILE :=
-        endif
-    endif
-endif
+TOOLCHAIN_LIST := riscv-none-elf-      \
+		  riscv32-unknown-elf- \
+		  riscv64-unknown-elf-
+
+VALID_TOOLCHAIN := $(foreach toolchain,$(TOOLCHAIN_LIST),     \
+		   $(shell which $(toolchain)gcc > /dev/null) \
+		   $(if $(filter 0,$(.SHELLSTATUS)),$(toolchain)))
 
+# Get the first element in valid toolchain list
+CROSS_COMPILE := $(word 1,$(VALID_TOOLCHAIN))
```

Comment:
You should not use `:=` since someone might override `CROSS_COMPILE` by setting this environment variable. That is, the logic should honor the value specified by `make  CROSS_COMPILE=riscv-none-elf-`.